pkg/database/dataValidation_test.go

```diff
@@ -0,0 +1,42 @@
+// Copyright Contributors to the Open Cluster Management project
+
+package database
+
+import (
+	"context"
+	"testing"
+
+	"github.com/doug-martin/goqu/v9"
+	pgx "github.com/jackc/pgx/v4"
+)
+
+func Test_ClusterTotals(t *testing.T) {
+	// Prepare a mock DAO instance
+	dao, mockPool := buildMockDAO(t)
+	clusterName := "cluster_foo"
+	batch := &pgx.Batch{}
+	// mock Result
+	br := BatchResults{
+		MockRows: MockRows{
+			mockData: []map[string]interface{}{{"count": 10}},
+			index:    1,
+		},
+	}
+	// mock queries
+	resourceCountSql, params, _ := goqu.From(goqu.S("search").Table("resources")).
+		Select(goqu.COUNT("*")).
+		Where(goqu.C("cluster").Eq(clusterName)).ToSQL()
+	batch.Queue(resourceCountSql, params)
+	edgeCountSql, params, _ := goqu.From(goqu.S("search").Table("edges")).
+		Select(goqu.COUNT("*")).
+		Where(goqu.C("cluster").Eq(clusterName),
+			goqu.C("edgetype").Neq("interCluster")).ToSQL()
+	batch.Queue(edgeCountSql, params)
+
+	mockPool.EXPECT().SendBatch(context.Background(), batch).Return(br)
```

Comment:
This block is duplicating the queries from the implementation. Can we mock these queries without goku? Using plain-text queries should simplify the test.